core/src/test/java/org/elasticsearch/monitor/process/ProcessProbeTests.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.monitor.process;
+
+import org.elasticsearch.bootstrap.Bootstrap;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.ElasticsearchTestCase;
+import org.junit.Test;
+
+import static org.elasticsearch.monitor.jvm.JvmInfo.jvmInfo;
+import static org.hamcrest.Matchers.*;
+
+public class ProcessProbeTests extends ElasticsearchTestCase {
+
+    ProcessProbe probe = ProcessProbe.getInstance();
+
+    @Test
+    public void testProcessInfo() {
+        ProcessInfo info = probe.processInfo();
+        assertNotNull(info);
+        assertThat(info.getRefreshInterval(), anyOf(equalTo(-1L), greaterThanOrEqualTo(0L)));
```

Comment:
maybe in a followup we can think about removing these -1s... see what platforms fail, and better fine-grain the stuff (e.g. add assumption for WINDOWS, IBM jdk, whatever it might be). Then we know when and where stats are available.